idaes/generic_models/properties/core/generic/generic_property.py

```diff
@@ -1808,6 +1812,97 @@ def rule_dens_mol_phase(b, p):
             raise
 
+    def _mole_frac_comp(self):
+        """If mole_frac_comp not state var assume mole_frac_phase_comp is"""
+        try:
+            def rule_mole_frac_comp(b, i):
+                return sum(b.phase_frac[p]*b.mole_frac_phase_comp[p, i]
+                    for p in b.params.phase_list)
+            self.mole_frac_comp = Expression(
+                    self.params.component_list,
+                    doc="Mole fraction of each component",
+                    rule=rule_mole_frac_comp)
+        except AttributeError:
+            self.del_component(self.mole_frac_comp)
+            raise
+
+    def _flow_mol_phase(self):
+        try:
+            def rule_flow_mol_phase(b, p):
+                return b.flow_mol*b.phase_frac[p]
+            self.flow_mol_phase = Expression(
+                    self.params.phase_list,
+                    doc="Molar flow rate of each phase",
+                    rule=rule_flow_mol_phase)
+        except AttributeError:
+            self.del_component(self.flow_mol_phase)
+            raise
+
+    def _flow_vol(self):
+        try:
+            self.flow_vol = Expression(
+                expr=self.dens_mol*self.flow_mol, doc="Volumetric flow rate")
+        except AttributeError:
+            self.del_component(self.flow_vol)
+            raise
+
+    def _flow_vol_phase(self):
+        try:
+            def rule_flow_vol_phase(b, p):
+                return b.flow_mol_phase[p]*b.dens_mol_phase[p]
+            self.flow_vol_phase = Expression(
+                    self.params.phase_list,
+                    doc="Volumetric flow rate of each phase",
+                    rule=rule_flow_vol_phase)
+        except AttributeError:
+            self.del_component(self.flow_vol_phase)
+            raise
+
+    def _flow_mass(self):
+        try:
+            self.flow_mass = Expression(
+                expr=self.mw*self.flow_mol, doc="Mass flow rate")
+        except AttributeError:
+            self.del_component(self.flow_mass)
+            raise
+
+    def _flow_mass_phase(self):
+        try:
+            def rule_flow_mass_phase(b, p):
+                return b.mw_phase[p]*b.flow_mol_phase[p]
+            self.flow_mass_phase = Expression(
+                    self.params.phase_list,
+                    doc="Mass flow rate of each phase",
+                    rule=rule_flow_mass_phase)
+        except AttributeError:
+            self.del_component(self.flow_mass_phase)
+            raise
+
+    def _flow_mol_comp(self):
+        try:
+            def rule_flow_mol_comp(b, i):
+                return self.mole_frac_comp[i]*self.flow_mol
+            self.flow_mol_comp = Expression(
+                self.params.component_list,
+                doc="Component molar flow rate",
+                rule=rule_flow_mol_comp)
+        except AttributeError:
+            self.del_component(self.flow_mol_comp)
+            raise
+
+    def _flow_mol_phase_comp(self):
```

Comment:
I would suggest removing this, and making it a standard expectation of the state definition module.